from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(821, 609)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(100)
        sizePolicy.setVerticalStretch(100)
        sizePolicy.setHeightForWidth(MainWindow.sizePolicy().hasHeightForWidth())
        MainWindow.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setPointSize(8)
        MainWindow.setFont(font)
        MainWindow.setAutoFillBackground(False)
        MainWindow.setStyleSheet("background-color: rgb(212, 237, 255);\n""color: rgb(25, 68, 124);")
        MainWindow.setTabShape(QtWidgets.QTabWidget.Rounded)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")

        # 'show list of all...' label
        self.list_label = QtWidgets.QLabel(self.centralwidget)
        self.list_label.setGeometry(QtCore.QRect(60, 329, 321, 41))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(10)
        self.list_label.setFont(font)
        self.list_label.setScaledContents(True)
        self.list_label.setAlignment(QtCore.Qt.AlignCenter)
        self.list_label.setObjectName("list_label")

        # show all cuisines button
        self.cuisines_pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.cuisines_pushButton.setGeometry(QtCore.QRect(60, 490, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        self.cuisines_pushButton.setFont(font)
        self.cuisines_pushButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.cuisines_pushButton.setStyleSheet("border-radius: 5px;\n""background-color: rgb(141, 182, 217);\n""color: rgb(237, 252, 255)")
        self.cuisines_pushButton.setObjectName("cuisines_pushButton")

        # show all categories button
        self.categories_pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.categories_pushButton.setGeometry(QtCore.QRect(60, 370, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        self.categories_pushButton.setFont(font)
        self.categories_pushButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.categories_pushButton.setStyleSheet("border-radius: 5px;\n""background-color: rgb(141, 182, 217);\n""color: rgb(237, 252, 255)")
        self.categories_pushButton.setObjectName("categories_pushButton")

        # show all ingredients button
        self.ingredients_pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.ingredients_pushButton.setGeometry(QtCore.QRect(60, 430, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        self.ingredients_pushButton.setFont(font)
        self.ingredients_pushButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.ingredients_pushButton.setStyleSheet("border-radius: 5px;\n""background-color: rgb(141, 182, 217);\n""color: rgb(237, 252, 255)")
        self.ingredients_pushButton.setObjectName("ingredients_pushButton")

        # login button
        self.login_pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.login_pushButton.setGeometry(QtCore.QRect(450, 400, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        self.login_pushButton.setFont(font)
        self.login_pushButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.login_pushButton.setAcceptDrops(False)
        self.login_pushButton.setStyleSheet("border-radius: 5px;\n""background-color: rgb(141, 182, 217);\n""color: rgb(237, 252, 255)")
        self.login_pushButton.setAutoRepeat(False)
        self.login_pushButton.setAutoDefault(False)
        self.login_pushButton.setDefault(False)
        self.login_pushButton.setFlat(False)
        self.login_pushButton.setObjectName("login_pushButton")

        # create new account button
        self.new_pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.new_pushButton.setGeometry(QtCore.QRect(450, 460, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        self.new_pushButton.setFont(font)
        self.new_pushButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.new_pushButton.setStyleSheet("border-radius: 5px;\n""background-color: rgb(141, 182, 217);\n""color: rgb(237, 252, 255)")
        self.new_pushButton.setObjectName("new_pushButton")

        # page title label
        self.title_label = QtWidgets.QLabel(self.centralwidget)
        self.title_label.setGeometry(QtCore.QRect(10, 24, 801, 111))
        font = QtGui.QFont()
        font.setFamily("Ink Free")
        font.setPointSize(42)
        self.title_label.setFont(font)
        self.title_label.setAutoFillBackground(False)
        self.title_label.setStyleSheet("")
        self.title_label.setFrameShape(QtWidgets.QFrame.VLine)
        self.title_label.setFrameShadow(QtWidgets.QFrame.Raised)
        self.title_label.setLineWidth(0)
        self.title_label.setScaledContents(True)
        self.title_label.setAlignment(QtCore.Qt.AlignCenter)
        self.title_label.setObjectName("title_label")

        # password entry
        self.password_lineEdit = QtWidgets.QLineEdit(self.centralwidget)
        self.password_lineEdit.setGeometry(QtCore.QRect(450, 290, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(10)
        self.password_lineEdit.setFont(font)
        self.password_lineEdit.setStyleSheet("background-color: qlineargradient(spread:pad, x1:0, y1:0.477, x2:1, y2:0.511, stop:0 rgb(239, 250, 255), stop:1 rgb(243, 247, 255));\n""border: 1px solid rgb(69, 111, 179);\n""border-radius: 5px;\n""\n""\n""")
        self.password_lineEdit.setFrame(False)
        self.password_lineEdit.setEchoMode(QtWidgets.QLineEdit.Password)
        self.password_lineEdit.setAlignment(QtCore.Qt.AlignCenter)
        self.password_lineEdit.setClearButtonEnabled(True)
        self.password_lineEdit.setObjectName("password_lineEdit")

        # login entry
        self.login_lineEdit = QtWidgets.QLineEdit(self.centralwidget)
        self.login_lineEdit.setGeometry(QtCore.QRect(450, 240, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(10)
        self.login_lineEdit.setFont(font)
        self.login_lineEdit.setStyleSheet("background-color: qlineargradient(spread:pad, x1:0, y1:0.477, x2:1, y2:0.511, stop:0 rgb(239, 250, 255), stop:1 rgb(243, 247, 255));\n""border: 1px solid rgb(69, 111, 179);\n""border-radius: 5px;\n""\n""\n""")
        self.login_lineEdit.setFrame(False)
        self.login_lineEdit.setAlignment(QtCore.Qt.AlignCenter)
        self.login_lineEdit.setClearButtonEnabled(True)
        self.login_lineEdit.setObjectName("login_lineEdit")

        # show random recipe button
        self.random_pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.random_pushButton.setGeometry(QtCore.QRect(60, 260, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        self.random_pushButton.setFont(font)
        self.random_pushButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.random_pushButton.setStyleSheet("border-radius: 5px;\n""background-color: rgb(141, 182, 217);\n""color: rgb(237, 252, 255)")
        self.random_pushButton.setAutoDefault(False)
        self.random_pushButton.setFlat(False)
        self.random_pushButton.setObjectName("random_pushButton")

        # search entry
        self.search_lineEdit = QtWidgets.QLineEdit(self.centralwidget)
        self.search_lineEdit.setGeometry(QtCore.QRect(60, 190, 321, 31))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.search_lineEdit.sizePolicy().hasHeightForWidth())
        self.search_lineEdit.setSizePolicy(sizePolicy)
        self.search_lineEdit.setBaseSize(QtCore.QSize(0, 0))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(10)
        self.search_lineEdit.setFont(font)
        self.search_lineEdit.setToolTipDuration(0)
        self.search_lineEdit.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.search_lineEdit.setAutoFillBackground(False)
        self.search_lineEdit.setStyleSheet("background-color: qlineargradient(spread:pad, x1:0, y1:0.477, x2:1, y2:0.511, stop:0 rgb(239, 250, 255), stop:1 rgb(243, 247, 255));\n""border: 1px solid rgb(69, 111, 179);\n""border-radius: 5px;\n""\n""\n""")
        self.search_lineEdit.setFrame(False)
        self.search_lineEdit.setCursorPosition(0)
        self.search_lineEdit.setAlignment(QtCore.Qt.AlignCenter)
        self.search_lineEdit.setDragEnabled(False)
        self.search_lineEdit.setClearButtonEnabled(True)
        self.search_lineEdit.setObjectName("search_lineEdit")

        # search label
        self.search_label = QtWidgets.QLabel(self.centralwidget)
        self.search_label.setGeometry(QtCore.QRect(60, 159, 321, 31))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.search_label.sizePolicy().hasHeightForWidth())
        self.search_label.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(10)
        self.search_label.setFont(font)
        self.search_label.setFrameShadow(QtWidgets.QFrame.Plain)
        self.search_label.setScaledContents(True)
        self.search_label.setAlignment(QtCore.Qt.AlignCenter)
        self.search_label.setObjectName("search_label")

        # or label
        self.or_label = QtWidgets.QLabel(self.centralwidget)
        self.or_label.setGeometry(QtCore.QRect(450, 430, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(9)
        self.or_label.setFont(font)
        self.or_label.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.or_label.setScaledContents(True)
        self.or_label.setAlignment(QtCore.Qt.AlignCenter)
        self.or_label.setObjectName("or_label")

        # error label
        self.error_label = QtWidgets.QLabel(self.centralwidget)
        self.error_label.setGeometry(QtCore.QRect(60, 220, 321, 31))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.error_label.sizePolicy().hasHeightForWidth())
        self.error_label.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(8)
        self.error_label.setFont(font)
        self.error_label.setFrameShadow(QtWidgets.QFrame.Plain)
        self.error_label.setScaledContents(True)
        self.error_label.setAlignment(QtCore.Qt.AlignCenter)
        self.error_label.setObjectName("error_label")

        self.error_label.raise_()

        self.or_label.raise_()
        self.list_label.raise_()
        self.cuisines_pushButton.raise_()
        self.categories_pushButton.raise_()
        self.ingredients_pushButton.raise_()
        self.login_pushButton.raise_()
        self.new_pushButton.raise_()
        self.title_label.raise_()
        self.password_lineEdit.raise_()
        self.login_lineEdit.raise_()
        self.random_pushButton.raise_()
        self.search_lineEdit.raise_()
        self.search_label.raise_()
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 821, 22))
        self.menubar.setObjectName("menubar")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "Recipes"))
        self.list_label.setText(_translate("MainWindow", "Show list of all:"))
        self.cuisines_pushButton.setText(_translate("MainWindow", "Cuisines"))
        self.categories_pushButton.setText(_translate("MainWindow", "Categories"))
        self.ingredients_pushButton.setText(_translate("MainWindow", "Ingredients"))
        self.login_pushButton.setText(_translate("MainWindow", "Log in"))
        self.new_pushButton.setText(_translate("MainWindow", "Create new account"))
        self.title_label.setText(_translate("MainWindow", "Recipes"))
        self.password_lineEdit.setPlaceholderText(_translate("MainWindow", "Password"))
        self.login_lineEdit.setPlaceholderText(_translate("MainWindow", "Login"))
        self.random_pushButton.setText(_translate("MainWindow", "Go to random recipe"))
        self.search_lineEdit.setPlaceholderText(_translate("MainWindow", "Dish name..."))
        self.search_label.setText(_translate("MainWindow", "Search:"))
        self.or_label.setText(_translate("MainWindow", "Or"))
        self.error_label.setText(_translate("MainWindow", ""))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())
